src/exprs.h

```diff
@@ -17,6 +17,8 @@
 #define GAP_EXPRS_H
 
 #include "system.h"
+#include "vars.h"
+#include "code.h"
```

Comment:
Please keep all #includes strictly alphabetically sorted.

Also, the need to add those includes here now is the main reason I did not convert those macros so far, as I was trying hard to avoid inter-header dependencies, to cut down (re)compile times. So, I am a bit sad to see this added here, but of course there are good reasons (and I can try to get rid of them again in the future through careful refactoring ;-)

Reply:
I more feel these headers were "always needed" (as we were defining macros which couldn't be used), but we just didn't realise before.

Reply:
I point out .clang-format says `SortIncludes: false`, and we don't really have a coding guide written anywhere I know of. I will rearrange them, but I don't know how I could have known to sort them.

Reply:
The convention is this: all headers are sorted, *except* that inside of `foo.c`, we always include `foo.h` first. This is why we can't let clang-format sort them for us, otherwise I'd have enabled this. This ensures that headers don't have accidentally missing dependencies, i.e., no "to use foo.h you must always include it after bar.h". 

As to how you could have known about this: Well, I did not mean to say you should have known, I merely pointed it out so it could be fixed. Although of course in theory, you could also have remembered it from your review of PR #2000, where this was explained :-). But again, I did not really expect that.

I don't know where I could document such a rule, though *shrug*. And to be honest, even if I documented it -- would you or anybody check that document before making a PR ;-). People don't run clang-format all the time, too *shrug*. Anyway, if you have a good idea where to place this, I can write it up there!

Reply:
In another project I work on, we use the fact clang-format won't reorder headers if you leave a blank line between them to group headers into sets which can then be automatically ordered.

Reply:
fixed

Reply:
Ah, I didn't know that. Excellent, let's enable it then, we already separate that "primary" header from the others with a newline.